internal/cortex/querier/queryrange/query_bytes_fetched.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Cortex Authors.
+// Licensed under the Apache License 2.0.
+
+package queryrange
+
+import (
+	"strconv"
+)
+
+const QueryBytesFetchedHeaderName = "M3-Fetched-Bytes-Estimate"
+
+func sumQueryBytesFetched(responses ...Response) int {
+	result := 0
+	for _, resp := range responses {
+		for _, hdr := range resp.GetHeaders() {
+			if hdr.GetName() == QueryBytesFetchedHeaderName {
+				for _, v := range hdr.Values {
```

Comment:
this is inefficient, you should be able to extract value from the corresponding headers right?

Reply:
see https://pkg.go.dev/net/http#Header